src/dill/fee-distributor-v2/contracts/FeeDistributor.vy

```diff
@@ -431,12 +432,13 @@ def burn(_coin: address) -> bool:
     """
     assert _coin == self.token
     assert not self.is_killed
-
-    amount: uint256 = ERC20(_coin).balanceOf(msg.sender)
+    
+    amount: uint256 = ERC20(_coin).allowance(msg.sender, self)
```

Comment:
I believe this is wrong. It should stay `balanceOf` because if I've approved uint256-1 tokens for this contract calling `burn` will try transfering that amount from my wallet into the contract (which I obviously won't have).

The test case doesn't catch this because it's not approving a large amount: 
`coin_a.approve(fee_distributor.address, 10**18, {"from": bob})`

Reply:
I see, I did this considering the fact that it might fail if the whole balance is not approved but did not consider infinite approval. Will change this.

Reply:
I typically see allowance checks done on the frontend, and the contract can just assume that everything's been approved properly 😉